routes/api/v1/meals.js

```diff
@@ -0,0 +1,51 @@
+var express = require('express');
+var router = express.Router();
+var Meal = require('../../../models').Meal;
+var Food = require('../../../models').Food;
+var MealFoods = require('../../../models').MealFoods;
+
+
+const parseMeals = (meals) => {
+   return meals.map(getMealObjects)
+}
+
+//The async/await here for parseFoods turned out to be key to getting the array of foods
+var getMealObjects = async (meal) => {
+  var getMealComponents = { id: meal.id,
```

Comment:
So the async / await solves the problem of multi-layered promise-land in this case?

Reply:
Yeah. We still need the returns every step of the way, but this await is making sure that the value of foods: is an array and not a promise, because if it's a promise there, it seems impossible to get the array any other way.